src/python/pants/backend/jvm/tasks/jvmdoc_gen.py

```diff
@@ -111,17 +111,18 @@ def docable(target):
     if not targets:
       return
 
-    with self.invalidated(targets) as invalidation_check:
-      def find_jvmdoc_targets():
-        invalid_targets = set()
-        for vt in invalidation_check.invalid_vts:
-          invalid_targets.update(vt.targets)
-        return invalid_targets
-
-      jvmdoc_targets = list(find_jvmdoc_targets())
-      if self.combined:
-        self._generate_combined(jvmdoc_targets, create_jvmdoc_command)
-      else:
+    if self.combined:
```

Comment:
This if statement should still be inside the `invalidated` block probably... because if _nothing_ has changed, it should not re-run.

The `if self.open` block in `_generate_combined` should probably move outside (after) the `self.invalidated` block though, so that we always open the report.

Reply:
OIC, so what you're saying is that the code should check for invalidation across the entire graph for combined, but if it finds any invalidated targets, return all targets, not just the invalidated ones.

That makes sense. I'll update to reflect that, and move the `self.open` logic up the control tree, as well.